### ISSUE TITLE
L028: support aliases in ORDER/GROUP BY (single-table SELECT)

### ISSUE THREAD

**Issue by enote-kane** [sqlfluff/sqlfluff#3053]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Using a column alias in an `ORDER BY` or `GROUP BY` for a single table `SELECT` does not seem to be supported.

### Expected Behaviour

No error

### Observed Behaviour

```
L:   3 | P:  10 | L028 | Unqualified reference 'test' found in single table
                       | select.
L:   7 | P:  10 | L028 | Unqualified reference 'test' found in single table
                       | select.
```

### How to reproduce

```sql
SELECT my_complex_function(my_table.my_col) AS test
FROM my_table
ORDER BY test;

SELECT my_complex_function(my_table.my_col) AS test
FROM my_table
GROUP BY test;
```

### Dialect

postgres

### Version

sqlfluff, version 0.12.0

### Configuration

```ini
[sqlfluff]
nocolor = True
dialect = postgres
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
So here's the story. L028 has an exception to check alias's but only for dialects that allow alias's in `WHERE` clause (snowflake, redshift) -  a weird concept as for most SQL engines the `WHERE` clause is executed first, and then the `SELECT` clause comes into effect.

We could easily remove that restriction and just allow this for all dialects. That would allow them to work in `ORDER BY` and `GROUP BY` and those solve this issue.

The downside is it would allow them to work in `WHERE` clauses for all dialects which, as per above, it shouldn't. But not sure that's the intent of this rule and shouldn't be captured by another, separate rule, if we don't want to allow it.

The alternative is to check the parent and then have an allow list/disallow list of statement types but not sure that's scalable (unless it's just disallow `WHERE`)? Also we'd need to pass the context to the nested function.

So I'm tempted with the easy option, for reasons given above.

Thoughts @enote-kane / @barrywhart ?

**Comment by tunetheweb** [comment]
Decided to go ahead and allow aliases everywhere in #3062  (even though that also allows them in `WHERE` clauses as well, even if they might not be be supported.
